src/imagery/i.eodag/i.eodag.py

```diff
@@ -530,7 +531,91 @@ def dates_to_iso_format():
+              each tuple consists of a queryable value and an operator,
+              as the dictionary values.
+              Dict["queryable", List(Tuple("queryable_value", "operator"))]
+    :rtype: Dict[str, List(Tuple(str, str))]
+
+    """
+    VALID_OPERATORS = ["eq", "ne", "ge", "gt", "le", "lt"]
+    DEFAULT_OPERATOR = "eq"
+    query_dicts = {}
+    if query is None:
+        return query_dicts
+    for parameter in map(str.strip, options["query"].split(",")):
+        if parameter == "":
+            continue
+        try:
+            key, values = map(str.strip, parameter.split("="))
+        except Exception as e:
+            gs.debug(e)
+            gs.fatal(_("Queryable <{}> could not be parsed".format(parameter)))
+        if key == "start":
+            try:
+                start_date = normalize_time(values)
+                query_dicts["start"] = [(start_date, DEFAULT_OPERATOR)]
+            except Exception as e:
+                gs.debug(e)
+                gs.fatal(
+                    _(
+                        "Queryable <{}> could not be parsed\nDate must be ISO formated".format(
+                            parameter
+                        )
+                    )
+                )
+            continue
+        if key == "end":
+            try:
+                end_date = normalize_time(values)
+                query_dicts["end"] = [(end_date, DEFAULT_OPERATOR)]
+            except Exception as e:
+                gs.debug(e)
+                gs.fatal(
+                    _(
+                        "Queryable <{}> could not be parsed\nDate must be ISO formated".format(
+                            parameter
+                        )
+                    )
+                )
+            continue
+        operator = None
+        values_operators = []
+        for value in map(str.strip, values.split("|")):
+            if value == "":
+                continue
+            if value.find("\\") != -1:
+                try:
+                    value, operator = map(str.strip, value.split("\\"))
+                except:
+                    gs.fatal(
+                        _("Queryable <{}> could not be parsed\n".format(parameter))
+                    )
+                if operator not in VALID_OPERATORS:
+                    gs.fatal(
+                        _(
+                            "Invalid operator <{}> for queryable <{}>. Available operators {}".format(
+                                operator, key, VALID_OPERATORS
+                            )
+                        )
+                    )
+            try:
+                value = float(value)
+            except:
+                if value.lower() == "none" or value.lower() == "null":
+                    value = None
+            values_operators.append((value, operator))
+        query_dicts[key] = values_operators
+    return query_dicts
+
+
+def filter_result(search_result, geometry=None, queryables=None, **kwargs):
```

Comment:
Given that all product properties can be used for searching, they should probably be used in the search stage already to minimize search time... But you can leave that for another day, as I guess there are som caveats e.g. when searching for data publised in a given time-frame...

Reply:
If you are talking about the queryables...
The searching parameters is inconsistent among the providers, see below:
https://eodag.readthedocs.io/en/stable/add_provider.html#opensearch-parameters-csv

So filtering, after done with searching, makes things less error prone... and AFAIK queryables are intended for filtering the data with the EODAG crunch after the search.